broker/machinebroker/server/machineclass_list.go

```diff
@@ -56,6 +57,18 @@ func (s *Server) gatherAvailableMachineClassNames(onmetalMachinePools []computev
 	return res
 }
 
+func (s *Server) gatherMachineClassQuantity(onmetalMachinePools []computev1alpha1.MachinePool) map[string]int64 {
```

Comment:
Please accumulate into `map[string]*resource.Quantity` and do the 'cast' to `int64` only at the last moment possible. In case we want to modify the `ori` interface to allow more precise resource quantities, this will make things a lot simpler + you always have the versatility of the `resource.Quantity` type.